### PR TITLE
Week 6_복습과제_조주현

### PR DESCRIPTION
**Pull request by freylia**

### CODE REVIEW

**Comment by chasubeen** [comment]
### **1-4. 시각화**
- 코드는 알맞게 작성되었으나, indentation으로 인해 조건이 반복문을 통해 제대로 걸리지 못해, 맨 마지막 RAD에 대한 시각화만 진행된 것 같습니다.
- regplot 그리는 코드를 반복문 안에 넣어주세요!

```
# 각 칼럼이 회귀 결과에 미치는 영향이 어느 정도인지 시각화해보기.

# 2개의 행과 4개의 열을 가진 Subplots를 이용.
# axs는 4x2개의 ax를 가짐.

fig, axs=plt.subplots(figsize=(16, 8), ncols=4, nrows=2)
lm_features =['RM', 'ZN', 'INDUS', 'NOX', 'AGE', 'PTRATIO', 'LSTAT', 'RAD']
for i, feature in enumerate(lm_features):
    row = int(i/4)
    col = i%4

    # 시본의 regplot을 이용해 산점도와 선형 회귀 직선을 함께 표현
    sns.regplot(x=feature, y='PRICE', data=bostonDF, ax=axs[row][col])
```

### **1-5.**
- 회귀 계수값의 의미를 설명해 달라는 의도로 출제된 문제였습니다.
- 예시
  - RM의 회귀 계수값은 8.5661034였다. 이는 PRICE와 RM이 양의 방향으로 선형성을 띈다는 것을 의미한다. 즉, 방의 크기가 클수록 주택 가격이 올라간다고 볼 수 있다.

### **5-1.**
- 현재 5번 문제에서는 당요병 데이터가 아닌 와인 데이터셋을 활용합니다.
- 따라서, train/test split 시 와인 데이터의 target을 활용하셔야 합니다.
  - 5-4번 문제에서 모델의 정확도가 낮게 나온 이유와도 연결됩니다.
- 답안
  ```
  from sklearn.preprocessing import StandardScaler
  from sklearn.model_selection import train_test_split

  scaler = StandardScaler()

  # 아래에 코드를 입력해주세요
  data_scaled = scaler.fit_transform(diabetesDF)

  # 괄호 안에 빈칸 채워주세요.
  # test_size=0.3
  X_train, X_test, y_train, y_test = train_test_split(data_scaled, y, test_size=0.3, random_state=0)
  ```

### **5-3**
- MSE와 R^2 또한 모델의 성능을 평가하는 지표이기는 하나, 해당 문제에서 요구한 성능 평가 metric은 '정확도'(accuracy)입니다.
- 따라서, 정확도를 활용하여 성능 평가를 진행해 주셔야 합니다.
- 답안
  ```
  from sklearn.metrics import accuracy_score

  print('accuracy: {0: .3f}'.format(accuracy_score(y_test, lr_preds)))
  ```

이번 과제도 수행하시느라 수고 많으셨습니다 😊
